evaluators/text/BleuEvaluator/__init__.py

```diff
@@ -7,17 +7,48 @@ class BleuEvaluator(BaseTextEvaluator):
     It will use the Bleu on NLTK package.
     A perfect match will score 1.0 and a perfect unmatch will score 0.0
     """
+
+    def count_ngram(self, text, max_ngram):
+        """Gets all n_grams from a given text
+            - text: the text to be analyzed
+            - max_ngram: max n_gram from the text
+            - counter: dictionary of n-gram and its count
+        """
+        from collections import Counter
+
+        counter = Counter()
+
+        for n_order in range(1, max_ngram):
+            for i in range(len(text) - n_order):
+                ngram = tuple(text[i:i+n_order])
+                counter[ngram] += 1
+        
+        return counter
+
     
-    def get_nltk_bleu_score(self, desired, actual):
+    def get_nltk_bleu_score(self, desired, actual, overlap):
         import nltk.translate.bleu_score as bleu    
         from nltk.translate.bleu_score import SmoothingFunction
 
-        return bleu.sentence_bleu(desired, actual, smoothing_function=SmoothingFunction().method4)
+        #Check the n-gram and reset the weights accordingly
+        overlaps = sum(overlap.values())
```

Comment:
is there any reference that explains this logic here?

Reply:
I added an article about it